integration-tests/utils/integration-utils.js

```diff
@@ -190,9 +190,28 @@ const dismissNotifications = async page => {
 }
 
 const signIntoTerra = async (page, { token, testUrl }) => {
-  !!testUrl && await page.goto(testUrl, waitUntilLoadedOrTimeout(60 * 1000))
-  await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout: 60 * 1000 })
-  await waitForNoSpinners(page)
+  if (!!testUrl) {
+    const timeout = 30 * 1000
+    const pageLoadedPromise = Promise.race([
+      page.waitForXPath('//*[@id="signInButton"]', { visible: true, timeout }),
+      page.waitForXPath('//a[@href="#workspaces"]', { visible: true, timeout })
+    ])
+    const loadPagePromise = Promise.all([
+      page.goto(testUrl, waitUntilLoadedOrTimeout(timeout)),
+      pageLoadedPromise
+    ])
+      .then(() => waitForNoSpinners(page))
+
+    try {
+      await loadPagePromise
+    } catch (err) {
+      console.error(err)
+      console.error(`Error: Page loading timed out during sign in. Reloading URL: "${testUrl}"`)
+      await page.reload({ waitUntil: 'load' })
+      await loadPagePromise
```

Comment:
Does this not re-throw the error, since the catch block would only run if `loadPagePromise` is rejected?

Reply:
Not necessarily. IIUC the goal is to try reloading if loading failed previously. If the reloading succeeds, the test would proceed with subsequent steps.

Reply:
Sorry, I don't fully understand your question. Error will be thrown if failing on the second attempt (L211),  and that should fails (stops) the test. If both the first attempt (L206) and the second attempt (L211) have failed, it might be due to the same root cause.

Reply:
Line 211 is not making a second attempt though. It's awaiting a promise that's already settled (rejected).

Reply:
Why do you think L211 is waiting for a promise that's already done (L206)? I may have misunderstanding about async and promises. Would you help me understand your concern better by explain this over a google meet?

Reply:
Oh yea @nawatts is right. @aweng98 did you mean to do something like below?

```javascript
const loadHomePageAndGoToTestUrl = (page, testUrl) => {
  const timeout = 30 * 1000

  return Promise.all([
    page.goto(testUrl, waitUntilLoadedOrTimeout(timeout)),
    Promise.race([
      page.waitForXPath('//*[@id="signInButton"]', { visible: true, timeout }),
      page.waitForXPath('//a[@href="#workspaces"]', { visible: true, timeout })
    ])
  ])
    .then(() => waitForNoSpinners(page))
}

const signIntoTerra = async (page, { token, testUrl }) => {
  if (!!testUrl) {
    try {
      await loadHomePageAndGoToTestUrl(page, testUrl)
    } catch (err) {
      console.error(err)
      console.error(`Error: Page loading timed out during sign in. Reloading URL: "${testUrl}"`)
      await page.reload({ waitUntil: 'load' })
      await loadHomePageAndGoToTestUrl(page, testUrl)
    }
  }

  await page.evaluate(token => window.forceSignIn(token), token)
  await dismissNotifications(page)
  await waitForNoSpinners(page)
}
```

Reply:
A promise represents the future result of an asynchronous operation. Awaiting the promise does not run the operation, it simply waits for it to finish.

You can see this in the console with something like:
```js
const sleep = (t) => {
    return new Promise(resolve => {
        setTimeout(resolve, t * 1000)
    })
}

(async () => {
    const promise = sleep(3)
    
    console.time()
    await promise
    console.timeEnd()

    console.time()
    await promise
    console.timeEnd()
})()
```

The first `await promise` does not return until `sleep` is finished. When `await promise` is called again, `sleep` is not re-run.

Reply:
Agreed with @kyuksel, I suspect the intention here is:

```js
try {
  await doSomething()
} catch (err) {
  // handle error
  await doSomething()
}
```
vs
```js
const promise = doSomething()
try {
  await promise
} catch (err) {
  // handle error
  await promise
}
```

Reply:
Ooh, crap. I understood finally. `await loadPagePromise` call are missing parenthesis. Good Catch! I wish I saw this earlier.

Reply:
Fixed problem and took Kaan's code suggestion. Thank you all.